### PR TITLE
feat: landing page

### PR DESCRIPTION
**Pull request by zawan-ila**
### Summary

This PR adds support for the landing page. While the functionality is there, the styling needs to be improved. As of now, the landing page looks like this:-

landing-page

### CODE REVIEW

File: src/components/Landing/Graph.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    <defs>
+    <linearGradient id="grad" x1="0" y1="0" x2="0" y2="1">
+      <stop offset="5%" stopColor="green" stopOpacity={0.8}/>
+      <stop offset="75%" stopColor="green" stopOpacity={0.1}/>
+      <stop offset="95%" stopColor="green" stopOpacity={0}/>
+    </linearGradient>
+
+    </defs>
+      <Area type="monotone" dataKey="price" stroke="green" strokeWidth={2}  fill="url(#grad)" />
```
Comment (azanbinzahid) [inline src/components/Landing/Graph.jsx]: Please lint your code or use some auto linter

File: src/components/Landing/LandingPage.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import  "./LandingPage.css"
+import Graph from './Graph'
+import Overview from './Overview'
+import logo from '../../assets/logo.png'
+
+export default function LandingPage() {
+
+  // Just some dummy data for the graph
```
Comment (azanbinzahid) [inline src/components/Landing/LandingPage.jsx]: make a separate file and import/store dummy data from there.

File: src/components/Landing/Overview.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import "./Overview.css"
+
+export default function Overview() {
+
+  const [numBlocks, setNumBlocks] = useState(0)
+  const [numTransactions, setNumTransactions] = useState(0)
+
+  useEffect(() => {
+    axios.get('http://127.0.0.1:8000/blocks/length').then(resp => {
```
Comment (azanbinzahid) [inline src/components/Landing/Overview.jsx]: use env file to store urls or any other config variables

File: src/index.js [hunk 4]
```diff
@@ -3,12 +3,17 @@ import ReactDOM from 'react-dom/client';
 import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
+import {BrowserRouter ,Routes, Route}  from 'react-router-dom'
```
Comment (azanbinzahid) [inline src/index.js]: comma/spacing issue - will be fixed by linter

File: src/index.js [hunk 5]
```diff
@@ -3,12 +3,17 @@ import ReactDOM from 'react-dom/client';
...
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
+  <BrowserRouter>
+    <Routes>
+      <Route path="/" element={<App />} />
+      <Route path="blocks" element={<App />} />
+      <Route path="transactions" element={<App />} />
```
Comment (azanbinzahid) [inline src/index.js:14]: I guess the routes are temporary? will we have separate component for each route?

**Review by azanbinzahid — CHANGES_REQUESTED** [review]

File: src/components/Landing/Graph.jsx (hunk 1 above)
Comment (zawan-ila) [reply src/components/Landing/Graph.jsx]: done

File: src/components/Landing/LandingPage.jsx (hunk 2 above)
Comment (zawan-ila) [reply src/components/Landing/LandingPage.jsx]: done

File: src/components/Landing/Overview.jsx (hunk 3 above)
Comment (zawan-ila) [reply src/components/Landing/Overview.jsx]: done

File: src/index.js (hunk 4 above)
Comment (zawan-ila) [reply src/index.js]: done

File: src/index.js (hunk 5 above)
Comment (zawan-ila) [reply src/index.js:14]: Except the first route others are temporary. We will fix them along the way(i.e when we add support for them)

File: .env [hunk 6]
```diff
@@ -0,0 +1,2 @@
+REACT_APP_BLOCKS_LENGTH_URL = 'http://127.0.0.1:8000/blocks/length'
+REACT_APP_TRANSACTIONS_LENGTH_URL = 'http://127.0.0.1:8000/transactions/length'
```
Comment (azanbinzahid) [inline .env]: store only base url and append path in code where it is used. You can use something like `BACKEND_URL` or `SERVICE_NAME_URL` as name.

File: src/components/Landing/LandingPage.css [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+.logo img{
+    height: 3rem;
+    margin-right: 0.8rem;
+}
+
+body{
+    background-color: rgb(219, 219, 219);
+}
```
Comment (azanbinzahid) [inline src/components/Landing/LandingPage.css]: add ending line

**Review by azanbinzahid — APPROVED** [review]

File: .env (hunk 6 above)
Comment (zawan-ila) [reply .env]: done

File: src/components/Landing/LandingPage.css (hunk 7 above)
Comment (zawan-ila) [reply src/components/Landing/LandingPage.css]: done

**Pull request merged**